timbermill-java/timbermill-local/src/main/java/com/datorama/oss/timbermill/ElasticsearchClient.java

```diff
@@ -942,11 +938,170 @@ long countByName(String name, String env) throws IOException {
     }
 
     public Bulker getBulker() {
-		return bulker;
-	}
+        return bulker;
+    }
+
+    public IndexRetryManager getRetryManager() {
+        return retryManager;
+    }
+
+    public void mergeIndexes() {
+        Set<String> currentAliases = getCurrentAliases();
+        currentAliases.forEach(alias -> {
+            try {
+                List<String> indicesToBeMoved = getIndicesToBeMoved(alias);
+                List<List<String>> partition = Lists.partition(indicesToBeMoved, 2);
+                partition.forEach(indexes -> {
+                    if (indexes.size() <= 1) {
+                        return;
+                    }
+                    String sourceIndex = indexes.get(0);
+                    String destIndex = indexes.get(1);
+                    moveIndexAnotherIndex(sourceIndex, destIndex);
+                });
+            } catch (Exception e) {
+                LOG.error("Failed to merge indexes for alias {}", alias, e);
+            }
+        });
+    }
+
+    private Set<String> getCurrentAliases() {
+        return getAliases(TIMBERMILL_INDEX_WILDCARD).keySet().stream().map(index -> index.replaceAll(INDEX_DELIMITER + "\\d+$", "")).collect(Collectors.toSet());
+    }
+
+    private List<String> getIndicesToBeMoved(String alias) throws IOException {
+        InputStream content = null;
+        try {
+            Set<String> indexNotToMerge = getAliases(alias + "*").keySet();
+            Request request = new Request("GET", "/_cat/indices/" + alias + "*?format=json");
+            Response response = client.getLowLevelClient().performRequest(request);
+            content = response.getEntity().getContent();
+            String json = IOUtils.toString(content);
+            JsonArray asJsonObject = new JsonParser().parse(json).getAsJsonArray();
+            List<String> indicesToMove = new ArrayList<>();
+            asJsonObject.forEach(jsonElement -> {
+                JsonObject jsonObject = jsonElement.getAsJsonObject();
+                String storeSize = jsonObject.get("store.size").getAsString();
+                String index = jsonObject.get("index").getAsString();
+                if (indexNotToMerge.contains(index)) {
+                    return;
+                }
+                if (isSizeLowerThen(storeSize, (long) (maxIndexSizeInGB * 0.1))) { // 10% of max index size
```

Comment:
probably better to extract to property (the 10%) so we will be able to further experiment with that later (and maybe increase it)

Reply:
done